last2libre/exporter.py

```diff
@@ -0,0 +1,237 @@
+        response = f.read()
+        f.close()
+        return response
+
+    def get_listen_list(self, response):
+        """Read XML page and get a list of listens and their info."""
+        xml_page = ET.fromstring(response)
+        listen_list = xml_page.getiterator('track')
+        return listen_list
+
+    def get_page_info(self, response):
+        """Check how many pages of listens user has."""
+        xml_page = ET.fromstring(response)
+        total_pages = xml_page.find(self.entity_type).attrib.get('totalPages')
+        return int(total_pages)
+
+    def parse_listen(self, listen):
+        """Extract info from every listen entry and output to list.
+
+        Args:
+            listen: String containing data for a single listen string to parse
+
+        Returns:
+            Sorted string of listen metadata.
+        """
+        if listen.find('artist').getchildren():
+            # Artist info nested in loved / banned tracks XML
+            artist_name = listen.find('artist').find('name').text
+            artist_mbid = listen.find('artist').find('mbid').text
+        else:
+            artist_name = listen.find('artist').text
+            artist_mbid = listen.find('artist').get('mbid')
+
+        if listen.find('album') is None:
+            # no album info for loved / banned tracks
+            album_name = ''
+            album_mbid = ''
+        else:
+            album_name = listen.find('album').text
+            album_mbid = listen.find('album').get('mbid')
+
+        track_name = listen.find('name').text
+        track_mbid = listen.find('mbid').text
+        date = listen.find('date').get('uts')
+
+        # Prepare final output from listen metadata
+        output = [date, track_name, artist_name, album_name,
+                  track_mbid, artist_mbid, album_mbid]
+
+        # Filter incorrect data
+        for i, v in enumerate(output):
+            if v is None:
+                output[i] = ''
+
+        return output
+
+    def write_listens(self, listens, fp):
+        """Write listens to an open file.
+
+        Args:
+            listens: List of listens (payload data)
+            fp: File pointer to open file object
+        """
+        for fields in listens:
+            fp.write(('\t'.join(fields) + '\n'))
+
+    def get_tracks(self):
+        current_page = self.page_number
+        response = self.connect_server()
+        total_pages = self.get_page_info(response)
+
+        if self.page_number > total_pages:
+            raise ValueError(
+                'First page ({}) is higher than total pages ({}).'.format(
+                    self.page_number, total_pages))
+
+        while current_page <= total_pages:
+            # Skip connect if on first page, already have that one stored.
+            if current_page > self.page_number:
+                response = self.connect_server()
+
+            listen_list = self.get_listen_list(response)
+            listens = []
```

Comment:
Since `dict()` is used a few times, it may be a good idea syntactically to use `list()` instead of the python2-style brackets.